server/service/src/processors/transfer/shipment/create_inbound_shipment.rs

```diff
@@ -89,6 +92,12 @@ impl ShipmentTransferProcessor for CreateInboundShipmentProcessor {
             &outbound_shipment,
             record_for_processing,
             request_requisition,
+            match outbound_shipment.invoice_row.r#type {
+                InvoiceRowType::OutboundShipment => InvoiceRowType::InboundShipment,
+                InvoiceRowType::OutboundReturn => InvoiceRowType::InboundReturn,
+                // TODO: is panicking here ok or should we return Ok(None)? (This should never happen bc of check above...)
```

Comment:
Maybe something like this would help https://github.com/msupply-foundation/open-msupply/compare/3191-generate-other-half...3191-generate-other-half(Example-of-intermediary-type) ?

Reply:
Ah yes! That's much nicer thank you :)